Code/Tools/TestImpactFramework/Runtime/Common/Code/Source/Artifact/Factory/TestImpactTestEnumerationSuiteFactory.cpp

```diff
@@ -84,4 +87,72 @@ namespace TestImpact
             return testSuites;
         }
     } // namespace GTest
+    namespace Python
+    {
+
+        AZStd::vector<AZStd::pair<AZStd::string, AZStd::vector<TestEnumerationSuite>>> TestEnumerationSuitesFactory(
+            const AZStd::string& testEnumerationData)
+        {
+            const char newLineDelim = '\n';
+            size_t startLine;
+            size_t endLine = 0;
+
+            AZStd::vector<AZStd::pair<AZStd::string, AZStd::vector<TestEnumerationSuite>>> pairs =
+                AZStd::vector<AZStd::pair<AZStd::string, AZStd::vector<TestEnumerationSuite>>>();
+            AZStd::map<AZStd::string, AZStd::map<AZStd::string, AZStd::set<AZStd::string>>> testMapping =
+                AZStd::map<AZStd::string, AZStd::map<AZStd::string, AZStd::set<AZStd::string>>>();
+
+            // Steps through our string line by line. Assumes lines are split by newline characters.
+            while ((startLine = testEnumerationData.find_first_not_of(newLineDelim, endLine)) != AZStd::string::npos)
+            {
+                endLine = testEnumerationData.find(newLineDelim, startLine);
+                AZStd::string curLine = testEnumerationData.substr(startLine, endLine - startLine);
+
+                curLine = curLine.substr(0, curLine.find_first_of('['));
+                // This regex matches pytest test names, in the form ScriptPath::TestFixture::TestName. It will accept both parameterized
+                // and unparamaterized tests (with or without [] at the end).
+                AZStd::basic_regex testNamePattern = AZStd::basic_regex("([a-zA-Z_/\\-0-9.]*\\.py)::([A-Za-z_/\\-0-9]*)::(.*)");
```

Comment:
Now that we're taking in the parameterized tests, this regex can be simplified to `"(.*)::(.*)::(.*)"`.

Reply:
Had to make a hybrid to account for some weird test name choices, but it has been simplified a little.